src/AdoNetCore.AseClient/AseCommand.cs

```diff
@@ -499,5 +502,28 @@ object ICloneable.Clone()
             return clone;
         }
 #endif
+
+        public string GetDataTypeName(int colindex)
+        {
+            var reader = ExecuteReader();
+            return reader.GetDataTypeName(colindex);
+        }
+
+        public void ResetCommandTimeout()
+        {
+            CommandTimeout = DefaultCommandTimeout;
+        }
+
+        public XmlReader ExecuteXmlReader()
```

Comment:
I thought ASE didn’t support XML...

Reply:
It doesn't AFAIK. In this case, the behaviour is actually to take a string-typed result and treat it as xml (emit an `XmlReader`)

Reply:
ok. Give some thought into how we’d add support for true XML types down the track if SAP adds into to the server. Can we switch on the data type and identify when to read native xml and when to coerce a string type into xml? Probably alright. We did something like this for getting guids from binary column types.